functests/dpdk/dpdk.go

```diff
@@ -997,14 +1019,51 @@ func waitForSRIOVStable() {
 	}, waitingTime, 1*time.Second).Should(BeTrue())
 }
 
-func filterPerformanceProfilesWithAvailableNodes(performanceProfiles []*perfv1alpha1.PerformanceProfile, nodeSelector map[string]string) []*perfv1alpha1.PerformanceProfile {
-	availableProfiles := make([]*perfv1alpha1.PerformanceProfile, 0)
+// findPerformanceProfileAndPolicyWithAvailableNodes finds a profile/sriovPolicy match for which a node with
+// allocatable resources is available. It will return a profile/sriovPolicy for a policy with resource name
+// "dpdknic", or a pair with the most available resource on node
+func findPerformanceProfileAndPolicyWithAvailableNodes(performanceProfiles []*perfv1alpha1.PerformanceProfile, nodeSelector map[string]string,
+) (preferredProfile *perfv1alpha1.PerformanceProfile, preferredResource string, preferredDevice *sriovv1.InterfaceExt, err error) {
+	currentResourceCount := 0
+	sriovInfos, err := sriovcluster.DiscoverSriov(sriovclient, SRIOV_OPERATOR_NAMESPACE)
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
Either expecting inside the function or returning the error is fine, but I'd try to be consistent in following one single approach.

Reply:
Done